### ISSUE TITLE
Clean validated data

### ISSUE THREAD

**Issue by jede** [ansman/validate.js#43]
It would be great to use validate to clean the data a user sends in. Basically I want a way to whitelist attributes to make them optional. So for a user you should be able to pass in description, but not createdAt. Here is a example of how it could work:

```coffee
data = 
  name: 'Test'
  description: 'Yaay'
  createdAt: 'omgomg'
 
constraints =
  name:
    presence: true
  description: true
 
validate.async(data, constraints).then (params) =>
  # params won't contain createdAt
  User.create(params)
```

**Comment by ansman** [comment]
Great idea.

My proposal is the following:
  * `validate.async` passes the attributes to the resolve callback.
  * `validate.async` cleans the data unless a new `cleanAttributes` is set to `false`
  * `validate.cleanAttributes` will be added that accepts an object with the attributes to keep (will also accept constraints)

Sounds resonable?

**Comment by jede** [comment]
Sounds amazing! :+1:

**Comment by ansman** [comment]
There we go, though I'm not sure I think clean attributes should be on per default since it can break backward compatibility, what do you think @jede

**Comment by jede** [comment]
Not sure. If the output from async previously was not defined, then I think its ok!

**Comment by ansman** [comment]
@jede It passed the attributed before as well

**Comment by ansman** [comment]
Released in 0.8.0 (42e5cb3451dbd39d45a13e200fc6b3e199c69b77)
